demo/src/main/java/com/novoda/merlin/demo/presentation/base/MerlinActivity.java

```diff
@@ -50,6 +50,12 @@ protected void onStop() {
         Logger.detach(logHandle);
     }
 
+    @Override
+    protected void onDestroy() {
+        merlin = null;
```

Comment:
The `Merlin` demo registers `Connectable`, `Disconnectable` and `Bindable`s on the `MerlinAcitivty`. These `Registerables` are stored internally in `Merlin` in a `List<>`. Unfortunately, this leads to a circular dependency, so unless we null-out the `Merlin` instance we will be leaking the `Activity`.

Removing these `Registerables` from `Merlin` does not resolve the issue because we are also passing and storing the `Activity` context in the `MerlinServiceBinder`. This issue has been present in the codebase for some time, having rolled back to check, but it should be addressed ASAP.